Other/SelectableCollection.cs

```diff
@@ -0,0 +1,194 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Text;
+using System.Windows;
+
+namespace DolphinDynamicInputTextureCreator.Other
+{
+    /// <summary>
+    /// specify the position in the list to be selected
+    /// </summary>
+    public enum SelectPropertie { Keep = default , First, Last, Previous, Next, Null}   
+
+    public class SelectableCollection<T> : ObservableCollection<T>, Interfaces.ISelectableCollection<T>
```

Comment:
I'm curious about this selection enumeration.  I would think the user would expect the behavior to be the same throughout the program.  Can you provide some examples where we'd want different behavior?

Without your input my thought would be remove should always keep, add should always do last, etc.

Reply:
you can also set the selection with it, so we could also get to the next texture with one button. so we can also change it at will.

Reply:
Hmm, I'm not sure I follow?  What do you mean you can set the selection with it?  The enumeration just says _how_ to select an item.

It feels like if we wanted a next or previous item, it should just be a function (`Next()`, `Previous()`) that sets the selected item, callable as a command.  This selection property gives us very little because our use cases won't be dynamic.